docs/components.md

```diff
@@ -364,6 +247,90 @@ m(ComponentUsingThis, {text: "Hello"})
 
 Be aware that when using ES5 functions, the value of `this` in nested anonymous functions is not the component instance. There are two recommended ways to get around this Javascript limitation, use ES6 arrow functions, or if ES6 is not available, use `vnode.state`.
 
+---
+
+### ES6 classes
+
+If it suits your needs (f.e. in object-oriented projects), components can also be written using ES6 class syntax:
```

Comment:
Minor nit: could you use "like" instead of "f.e." here? The latter isn't really a common abbreviation even among native speakers, so people might not necessarily know what it means.

```suggestion
If it suits your needs (like in object-oriented projects), components can also be written using ES6 class syntax:
```

Reply:
Heh, agreed. Nice catch :)